### ISSUE TITLE
Learning the initial weights of the heads leads to NaN

### ISSUE THREAD

**Issue by tristandeleu** [snipsco/ntm-lasagne#2]
When setting the `learn_init=True` parameter on the heads, the error and parameters of the heads become NaN after a few iterations (not necessarily on the first one, but it can happen after 100+ iterations).

How to reproduce it:

``` python
heads = [
    WriteHead([controller, memory], shifts=(-1, 1), name='write', learn_init=True),
    ReadHead([controller, memory], shifts=(-1, 1), name='read', learn_init=True)
]
```

This is a non-blocking issue since learning these weights may not actually make sense (we can just leave equi-probability as is as the first step).

**Comment by EderSantana** [comment]
hi @tristandeleu, when learning the initial weights, are you making sure they are behind of a softmax? In other word, are you learning the initial logits instead? If so, there is no problem if they get negative values. I had this problem in my NTM implementation as well.

**Comment by tristandeleu** [comment]
When I originally opened this issue I didn't, which was a mistake on my end. I haven't tried to learn the logit. But indeed you're right, I think this is the right solution (I only sketched the idea in this issue).
All in all I ended up leaving the `learn_init=False` for the weights in my experiments, and initialize them as one-hot vectors. But I haven't found a good way to allow both to fix the initialization (eg. with `OneHot`) and learning logit weights.

**Comment by EderSantana** [comment]
I'm new to your codebase, can you point me out where you get the initial weights, I could try to check that out.

**Comment by tristandeleu** [comment]
The initial weights are defined here: https://github.com/snipsco/ntm-lasagne/blob/master/ntm/heads.py#L102
But for now, there's no correct way to learn these weights unfortunately.
